js/screens/DiscoverScreen.js

```diff
@@ -0,0 +1,431 @@
+          // per https://developer.apple.com/documentation/usernotifications/asking_permission_to_use_notifications?language=objc
+          // only the first authorization request is honored
+          Linking.openURL('App-Prefs:NOTIFICATIONS_ID');
+        }
+      }}
+      contentContainerStyle={{paddingHorizontal: 10}}
+      text1Style={{
+        fontSize: 17,
+        fontWeight: '400',
+      }}
+      text2Style={{
+        fontSize: 17,
+      }}
+    />
+  ),
+};
+
+class DiscoverScreen extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.defaultSearch = '#locale-en';
+
+    this.state = {
+      loading: true,
+      page: 1,
+      results: [],
+      selectionCount: 0,
+      term: '',
+      selectedTag: '',
+      hasMoreResults: false,
+      progress: 0.5,
+      largerUI: props.screenProps.largerUI,
+    };
+
+    this._siteManager = this.props.screenProps.siteManager;
+    this.baseUrl = 'https://discover.discourse.org/search.json?q=';
+
+    this.debouncedSearch = debounce(this.doSearch, 750);
+
+    this.doSearch('');
+  }
+
+  doSearch(term, opts = {}) {
+    const searchTerm = term === '' ? this.defaultSearch : term;
+    const order = term.startsWith('order:') ? '' : 'order:featured';
+    const searchString = `#discover ${searchTerm} ${order}`;
+    const q = `${this.baseUrl}${encodeURIComponent(searchString)}&page=${
+      opts.pageNumber || 1
+    }`;
+
+    fetch(q)
+      .then(res => res.json())
+      .then(json => {
+        if (json.topics) {
+          this.setState({
+            results: opts.append
+              ? this.state.results.concat(json.topics)
+              : json.topics,
+            loading: false,
+            hasMoreResults:
+              json.grouped_search_result &&
+              json.grouped_search_result.more_full_page_results,
+          });
+        } else {
+          if (opts.pageNumber > 1) {
+            // Skip early if page number filter is somehow returning no results
+            // otherwise, this can cause a reset when scrolling
+            // TODO: handle this situation a little better maybe?
+            return;
+          }
+
+          this.setState({results: [], loading: false, hasMoreResults: false});
+          Site.fromTerm(term)
+            .then(site => {
+              if (site) {
+                this.setState({
+                  results: [
+                    {
+                      ...site,
+                      featured_link: site.url,
+                      excerpt: site.description,
+                      discover_entry_logo_url: site.icon,
+                    },
+                  ],
+                });
+              }
+            })
+            .catch(e => {
+              console.log(e);
+            });
+        }
+      })
+      .catch(e => {
+        console.log(e);
+      });
+  }
+
+  addSite(term, opts = {}) {
+    if (term.length === 0) {
+      return new Promise((resolve, reject) => reject());
```

Comment:
Ideally all of it would consistently use `async/await` 😅

Reply:
Yeah, true, this was moved here from a different place and I wanted to keep the changes to a minimum.